Add new release blog for v0.25 (#4152)
knative/docs: blog/releases/announcing-knative-v0-25-release.md

- Trigger ready status now takes into account the `DeadLetterURI`
- The `kn` CLI now looks for plugin in the `$PATH` allowing for easy installation of plugins like `kn-quickstart`

### Serving v0.25

<!-- Original notes are here: https://github.com/knative/serving/releases/tag/v0.25.0 -->

#### 🚨 Breaking or Notable Changes

- **Renaming of some `net-kourier` component**

    Related issue: [knative/networking#448](https://github.com/knative/networking/issues/448)

    As part of our efforts to GA/1.0 we've standardized on the naming of our networking plugins that
    are installed alongside Serving. If you're managing your Knative deployment manually with
    `kubectl` **this will require a two-phase upgrade process**. In order to upgrade [net-kourier to v0.25.0](https://github.com/knative-sandbox/net-kourier/releases/tag/v0.25.0) using `kubectl` please follow the steps:

        ```bash
        # Apply the new release
        $ kubectl apply -f net-kourier.yaml

        # Once the deployment is ready apply the same file but
        # prune the old resources

>         $ kubectl apply -f net-kourier.yaml \

Reviewer: where do I get or download this net-kourier.yaml? Perhaps it can be added to the paragraph before or use a URL

Author: Hi, I state above where to get the yaml https://github.com/knative-sandbox/net-kourier/releases/tag/v0.25.0 for `net-kourier.yaml` if you are upgrading then you are already familiar with the kourier installation since you did it once.

### Eventing v0.25

<!-- Original notes are here: https://github.com/knative/eventing/releases/tag/v0.25.1 -->

#### 💫 New Features & Changes

- Add `DeadLetterURI` to Trigger status, block ready until it is resolvable ([#5551](https://github.com/knative/eventing/pull/5551))
- Add a health endpoint to `event_display` for enabling readiness probes. ([#5608](https://github.com/knative/eventing/pull/5608))
- Adds support for HTTP OPTIONS and CloudEvents Webhook preflight ([#5542](https://github.com/knative/eventing/pull/5542))

#### 🐞 Bug Fixes

- PingSource schedules starting with `@every` are now properly rejected. ([#5585](https://github.com/knative/eventing/pull/5585))

### Eventing Extensions

#### Apache Kafka Broker v0.25

<!-- Original notes are here: https://github.com/knative-sandbox/eventing-kafka-broker/releases/tag/v0.25.0 -->

#### 💫 New Features & Changes

- The Trigger status contains the resolved URL of the configured dead letter sink. ([#1092](https://github.com/knative-sandbox/eventing-kafka-broker/pull/1092))

#### 🐞 Bug Fixes

> - Fix to support subscribers that are Kubernete service, service's endpoint doesn't contain a trailing slash. ([#1123](https://github.com/knative-sandbox/eventing-kafka-broker/pull/1123))

Reviewer 2: ```suggestion
- Fix to support subscribers that are a Kubernetes service. The service's endpoint doesn't contain a trailing slash. ([#1123](https://github.com/knative-sandbox/eventing-kafka-broker/pull/1123))
```

#### Apache Kafka Source and Channels v0.25

<!-- Original notes are here: https://github.com/knative-sandbox/eventing-kafka/releases/tag/v0.25.0 -->

#### 💫 New Features & Changes

- Distributed KafkaChannel Dispatcher will now re-queue KafkaChannels after failed reconciliation which will improve various failure recovery scenarios, but could consume CPU resources if reconciliation is blocked by underlying system issues (bad Kafka Secret config, etc.) ([#795](https://github.com/knative-sandbox/eventing-kafka/pull/795))

> - Separated CRDs to extra, additional yaml manifest (similar to eventing) ([#799](https://github.com/knative-sandbox/eventing-kafka/pull/799))

Reviewer 2: ```suggestion
- Separated CRDs to extra, additional YAML manifest (similar to eventing) ([#799](https://github.com/knative-sandbox/eventing-kafka/pull/799))
```

#### 💫 New Features & Changes

- The new plugin `kn-quickstart` is now part of the home-brew plugins suite. Install the plugin using `brew install knative-sandbox/kn-plugins/quickstart` then use it with `kn quickstart kind` this will create a kind cluster with knative installed. Make sure to update `kn` to `v0.25` for example using `brew upgrade kn`

- Deprecate `lookup-path` as path lookup will always be enabled in the future ([#1422](https://github.com/knative/client/pull/1422))
- Add `--tls` option to domain create command ([#1419](https://github.com/knative/client/pull/1419))
- Lookup plugins in `$PATH` by default ([#1412](https://github.com/knative/client/pull/1412))
- Add `--class` flag to broker create command ([#1402](https://github.com/knative/client/pull/1402))
- Add `darwin/arm64` support to kn ([#1401](https://github.com/knative/client/pull/1401))
- Add `base64` data handling to Ping commands ([#1392](https://github.com/knative/client/pull/1392)) ([#1388](https://github.com/knative/client/pull/1388))
- Add support for multiple containers in Service spec ([#1382](https://github.com/knative/client/pull/1382))

> - Make `--cmd` flag as an array instead of string ([#1380](https://github.com/knative/client/pull/1380))

Reviewer 2: ```suggestion
- Make `--cmd` flag an array instead of string ([#1380](https://github.com/knative/client/pull/1380))
```

> - Add an `client.knative.dev/updateTimestamp` annotation to trigger a new revision when required ([#1364](https://github.com/knative/client/pull/1364))

Reviewer 2: ```suggestion
- Add a `client.knative.dev/updateTimestamp` annotation to trigger a new revision when required ([#1364](https://github.com/knative/client/pull/1364))
```
